keras_trainer/trainer.py

```diff
@@ -143,8 +108,9 @@ def run(self):
                     expanded_model.layers[i].set_weights(self.model.layers[i].get_weights())
                 self.model = expanded_model
             else:
+                # Input Shape = None to be able to process arbitrary input size images
                 self.model = self.model_spec.klass(
-                    input_shape=self.model_spec.target_size,
+                    input_shape=None,
```

Comment:
Some models will still use an input shape / target size though, doesn't this remove that ability? I would keep it as `self.model_spec.target_size` and allow the user to pass in a `None` target size if they want arbitrary sizes.

Reply:
Also the fact that a test didn't fail from this change means we're missing a test that covers that, could you add one before merging?

Reply:
But the same variable is used to configure image load size (https://github.com/triagemd/keras-trainer/blob/freeze_layers/keras_trainer/trainer.py#L188-L205).

Keras constraints input images to be the size defined here:

```
self.model = self.model_spec.klass(
-                    input_shape=self.model_spec.target_size,
+                    input_shape=None,
```

By setting Input Shape to None I am saying that the model can accept arbitrary Input shape, while being trained with `self.model_spec.target_size` size images. (So the only difference would be that we can accept all image sizes now)

We can make an input shape variable if you want for setting the input shape of the models. (I wouldn't touch the one in model specs)

Reply:
Maybe I'm not following but it feels like we're creating an inconsistency here. If you pass in `None` as input shape to most models they're going to use their default like here https://github.com/keras-team/keras/blob/master/keras/applications/densenet.py#L180 — so if the user is specifying an explicit input shape / target size via the model spec then it's going to be inconsistent.

Reply:
As most of the models are fully convolutional now, with global average/max pooling layers at the end, CNNs can process variable size inputs. 
I don't find useful to restrict the input to only a determined shape, but I can add it as option if you find that appropriate.

Reply:
Yea I'd keep backwards compatibility and allow a given input shape so that if the user assigns one then it matches their expectations when training too.

Reply:
You are right, it will return the networks default, but only if `include_top=True`.
If not, it sets it as (None, None, 3). 
What about I make a variable and if it is inputed by the user it overrides model_specs_target size?

Reply:
@jsalbert https://github.com/triagemd/keras-trainer/pull/18/commits/1def26254ecb953378cdb24cb959b9ce82174daf looks good :+1: